### PR TITLE
new config var enable_key_logging added, request logging unified

### PR DESCRIPTION
**Pull request by dencoded**
added changes to fix https://github.com/TykTechnologies/tyk/issues/1203

1. added helper function to log request info and have more grain control about what to put in log for request
2. added new config parameter `enable_key_logging` with default value false which will switch on/off logging for tokens/keys

I tried to to catch all cases where we log info about http-request and changed them to use new helper function `getLogEntryForRequest`. It prepares logrus.Entry with passed data but leaves freedom to decide about message log-level to caller. If we go with this approach the assumption is to keep eye on how we log things and use this function to make sure sensitive info won't be logged.

### CODE REVIEW

**Comment by buger** [comment]
See the error in circleCI it fails on `goimports` check.

File: log_helpers.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+func getLogEntryForRequest(r *http.Request, key string, data map[string]interface{}) *logrus.Entry {
+	// populate http request fields
+	fields := logrus.Fields{
+		"path":   r.URL.Path,
+		"origin": requestIP(r),
+	}
+	// add key to log if configured to do so
+	if key != "" && config.Global.EnableKeyLogging {
+		fields["key"] = key
```
Comment (buger) [inline log_helpers.go]: Looks good! But I think it makes sense make it obvious to the user that tokes are hidden. Smth like this:
```
if key != "" {
    fields["key"] = key

    if !config.Global.EnableKeyLogging {
      key = "<hidden>"      
    }
}
```

**Comment by mvdan** [comment]
bump @buger

**Comment by buger** [comment]
👌

**Pull request merged**